pkg/kubecost/allocationfilter.go

```diff
@@ -213,6 +248,26 @@ func (filter AllocationFilterCondition) Matches(a *Allocation) bool {
 		} else {
 			log.Warnf("Allocation Filter: invalid 'contains' call for non-list filter value")
 		}
+	case FilterNotContains:
+		if stringSlice, ok := valueToCompare.([]string); ok {
+			// services!:"__unallocated__" should match
+			// len(a.Properties.Services) > 0
+			//
+			// TODO: is this true?
+			if filter.Value == UnallocatedSuffix {
+				return len(stringSlice) > 0
+			}
```

Comment:
I'm a little wary of `__unallocated__` queries and I don't know what the right answer is. What Allocations should `services!:"__unallocated__"` match? I will update the test cases accordingly once we come to consensus.

Reply:
What do you think about just matching the existing behavior?

```go
// special case: allocations that cannot get services or have no services
// pass the filter for `filterValue == "__unallocated__"`
if len(serviceList) == 0 {
	return (filterValue == "__unallocated__")
}
```

Reply:
As far as I know, we have never supported the idea of inequality with `"__unallocated__"` before. In the current implementation of `AllocationFilterCondition.Matches()`, we do something that looks like the code referenced in this review comment. See here:

https://github.com/kubecost/opencost/blob/040df8955b01af129898b9c980415c9553cd1ec2/pkg/kubecost/allocationfilter.go#L193-L197

The idea is that we have to change the `if` condition for `!=` queries because if its like this:
 ```go
if len(serviceList) > 0 { 
    return filterValue != "__unallocated__"
}
```

you'll incorrectly enter that block for non-unallocated queries.

This might be a little confusing, it's hard to express this properly. Let me know if this makes sense. Ideally we get this straight so the behavior is clear to others in the future.

Reply:
Resolves out of band -- we agree that the logic looks right as it stands.